packages/app/src/systems/Account/components/AccountList/AccountList.tsx

```diff
@@ -19,17 +20,20 @@ export function AccountList({
   onPress,
   onExport,
   onUpdate,
+  onToggleHidden,
 }: AccountListProps) {
   const [showHidden, setShowHidden] = useState(() => false);
   const [anyHiddenAccounts, setAnyHiddenAccounts] = useState(false);
+  const [isPrimaryAccount, setIsPrimaryAccount] = useState(false);
 
   function toggle() {
     setShowHidden((s) => !s);
   }
 
   useEffect(() => {
-    const hiddenAccounts = (accounts ?? []).some((acc) => acc.isHidden);
-    setAnyHiddenAccounts(hiddenAccounts);
+    const hiddenAccounts = (accounts ?? []).filter((acc) => acc.isHidden);
```

Comment:
I see that there are a couple of `accounts ?? []` in this component.  Imo it would be better to have some variable 
`const accounts = accounts ?? []` and then use that variable throughout the component.

Reply:
Also what are the cases for when accounts is undefined?  Would it be possible to change the type of the accounts prop to not accept undefined?

Reply:
> Also what are the cases for when accounts is undefined? Would it be possible to change the type of the accounts prop to not accept undefined?

We're getting `accounts` from the state machine, where the initial context is set to `undefined` instead of an empty array. @LuizAsFight @luizstacio can probably recommend if we'd like to update the default state of the `accountsMachine` to be an empty array instead of `undefined`

Reply:
this is fixed I guess? not needed `accounts ?? []` anymore as data is coming from  `useAccounts` hook